internal/github/languages.go

```diff
@@ -0,0 +1,112 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/pkg/errors"
+	"github.com/shurcooL/githubv4"
+)
+
+type Language struct {
+	Size  int
+	Name  string
+	Color string
+}
+
+type Languages []Language
+
+func (l Languages) String() string {
+	str := ""
+
+	for _, lang := range l {
+		str += fmt.Sprintf("**%s** %d bytes\n", lang.Name, lang.Size)
+	}
+
+	return str
+}
+
+func (g *GithubService) GetLanguagesByUsername(ctx context.Context, username string) (Languages, error) {
+
+	languageMap := map[string]*Language{}
+
+	var afterCursor string
+
+	for {
+
+		var languagesQuery struct {
+			User struct {
+				Repositories struct {
+					PageInfo struct {
+						HasNextPage bool
+						EndCursor   string
+					}
+					Nodes []struct {
+						ID        string
+						Name      string
+						Languages struct {
+							Edges []struct {
+								Size int
+								Node struct {
+									Color string
+									Name  string
+								}
+							}
+						} `graphql:"languages(first: 100, orderBy: {field: SIZE, direction: DESC})"` // If someone is using over 100 different languages in a repo, the results will be incorrect. I'm okay with this because you might be a monster if you have over 100 languages in a single repo.
+					}
+				} `graphql:"repositories(ownerAffiliations: OWNER, isFork: false, after: $after, first: 100)"`
+			} `graphql:"user(login: $username)"`
+		}
+
+		params := map[string]interface{}{
+			"username": githubv4.String(username),
+		}
+
+		if afterCursor == "" {
+			params["after"] = (*githubv4.String)(nil)
```

Comment:
We need this logic because graphql expects a nil value in place of the string to work correctly. We cannot pass in an empty string.